src/sentry/static/sentry/app/views/eventsV2/utils.tsx

```diff
@@ -335,7 +335,11 @@ export function downloadAsCsv(tableData, columnOrder, filename) {
     }),
   });
 
-  const encodedDataUrl = encodeURI(`data:text/csv;charset=utf8,${csvContent}`);
+  // Need to also manually replace # since encodeURI skips them
+  const encodedDataUrl = encodeURI(`data:text/csv;charset=utf8,${csvContent}`).replace(
```

Comment:
@wmak I think the better strategy would be to do 

```js
const encodedDataUrl = `data:text/csv;charset=utf8,${encodeURIComponent(csvContent)}`;
```

Essentially using `encodeURIComponent()` instead of `encodeURI()` as outlined in https://stackoverflow.com/questions/55267116/how-to-download-csv-using-a-href-with-a-number-sign-in-chrome

Reply:
:+1: to using the better browser function.

Reply:
Hm, this was actually my original solution here, but couldn't get `encodeURIComponent` to work cause its converting too much, since commas and newline get converted too which break the resulting CSV.
eg.
```
>> encodeURIComponent(",\r\n")
<< "%2C%0D%0A"
```
And even per row it works too hard cause it'll fiddle with our spaces:
eg.
Message: `Divided by Zero`
Message in CSV: `Divided%20by%20Zero`

Reply:
@wmak Hmmmm. It should work. Have you tried exporting the CSV data and introspecting it? 

The escape sequences generated by `encodeURIComponent()` is only for encoding the CSV data for the URL in the `href` component of the anchor tag.

Reply:
This was me being silly and running `encodeURI` ontop of `encodeURIComponent`, fixed now!